omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-EncounterPage.xml

```diff
@@ -294,10 +296,37 @@
                         isValid = false;
                     }
                 }
+               <!--Not to exceed number of booking for a specific return date. Default is 100 clients-->
+               <!-- formatting date String from YYYY-MM-DD to DD/MM/YYYY to pass in the Url-->             
+               var selectedDate = jq('#appointmentDate').find("input[type=hidden]").val();
+
+               var formattedDate = changeFieldDateToJavascriptDate(selectedDate);
+               var URL = window.location.origin + '/' + OPENMRS_CONTEXT_PATH + '/ws/rest/v1/ugandaemr/appointmentcount?nextAppointmentDate=' + formattedDate.replace(/['"]+/g, '') + '&amp;encounterType=8d5b2be0-c2cc-11de-8d13-0010c6dffd0f';
+
+               var numberOfAppointmentsOnSelectedDate = jq.ajax({
+                     type: 'GET',
+                     dataType: 'json',                   
+                     url: URL,
+                     async:false,
+                     success: function(response){
+                         if(response.appointmentcount &gt;= jq("#numberOfPatients").html()){
+                            getField('appointmentDate.error').html('Please enter another date because the selected date has reached a max limit').show();                       
+                            isValid = false;
+                         }
+                     },
+                    error: function(jqXHR, textStatus, errorThrown) {
+                           getField('appointmentDate.error').html('Unable to determine the number of already booked patients on this date').show();
```

Comment:
@reagan-meant add the date so that a user knows which date it is, and also add part of the textStatus for troubleshooting later

Reply:
Done